drivers/hmis/app/graphql/mutations/update_external_form_submission.rb

```diff
@@ -8,24 +8,71 @@ module Mutations
   class UpdateExternalFormSubmission < CleanBaseMutation
     argument :input, Types::HmisSchema::ExternalFormSubmissionInput, required: true
     argument :id, ID, required: true
+    argument :project_id, ID, required: false # not required for backwards compatibility
 
-    field :external_form_submission, Types::HmisSchema::ExternalFormSubmission, null: false
+    field :external_form_submission, Types::HmisSchema::ExternalFormSubmission, null: true
     field :errors, [Types::HmisSchema::ValidationError], null: false, resolver: Resolvers::ValidationErrors
 
-    def resolve(id:, input:)
+    def resolve(...)
+      Hmis::Hud::Base.transaction do
+        _resolve(...)
+      end
+    end
+
+    protected
+
+    def _resolve(id:, project_id:, input:)
       record = HmisExternalApis::ExternalForms::FormSubmission.find(id)
       raise 'Access denied' unless allowed?(permissions: [:can_manage_external_form_submissions])
 
       record.assign_attributes(**input.to_params)
 
-      errors = []
-      if record.valid?
-        record.save!
-      else
-        errors = record.errors
-        record = nil
+      errors = HmisErrors::Errors.new
+
+      unless record.valid?
+        errors.add_ar_errors(record.errors&.errors)
+        return { errors: errors }
       end
 
+      if record.status_changed? && record.status == 'reviewed' && !record.spam
+        definition = record.definition
+
+        # Only if there are Client and/or Enrollment fields in the form definition, initialize an enrollment
+        # (which will in turn initialize a Client, inside the form processor).
+        if definition.link_id_item_hash.values.find { |item| ['ENROLLMENT', 'CLIENT'].include?(item.mapping.record_type) }
+          project = Hmis::Hud::Project.find(project_id)
+          record.build_enrollment(project: project, data_source: project.data_source, entry_date: record.created_at)
+        end
+
+        form_processor = record.form_processor || record.build_form_processor(definition: definition)
+
+        form_processor.values = record.cleaned_values
+        form_processor.hud_values = input.hud_values
+
+        # Validate based on form definition
+        form_validations = form_processor.collect_form_validations
+        errors.push(*form_validations)
+
+        # Run to create CDEs, and client/enrollment if applicable
+        form_processor.run!(user: current_user)
+
+        # Collect validations and warnings from AR Validator classes
+        record_validations = form_processor.collect_record_validations(user: current_user)
+        errors.push(*record_validations)
+
+        errors.deduplicate!
+        return { errors: errors } if errors.any?
+
+        if record.enrollment&.new_record?
+          record.enrollment.client.save!
```

Comment:
This throws an error if the form values have any client info but lack relationship to HoH. But the user can't do anything about that validation at this stage -- it would be a problem with the form logic.
- do we expect the PIT form to have a question for relationship to HoH?
- if not, does the html form support hidden fields? maybe we could have it as a hidden field inferred from other values, such as the # of household members, which I believe will only be collected for the first HH member?

Reply:
Hmm. I think PIT should have a question for relationship, but not sure if it would be required. 

Another approach would be to default to `relationship_to_hoh: 99` in the EnrollmentProcessor, if the form values don't contain a relationship to hoh. We do something similar for VeteranStatus in the ClientProcessor, but there we still expect the key to be present. I think that could be a reasonable approach here

Reply:
Discussed - tentatively, assume it's present?

Reply:
Outdated, see https://github.com/greenriver/hmis-warehouse/pull/4667#discussion_r1744270493